internal/backup/volumesnapshot_action.go

```diff
@@ -54,17 +57,26 @@ func (p *VolumeSnapshotBackupItemAction) AppliesTo() (velero.ResourceSelector, e
 
 // Execute backs up a CSI volumesnapshot object and captures, as labels and annotations, information from its associated volumesnapshotcontents such as CSI driver name, storage snapshot handle
 // and namespace and name of the snapshot delete secret, if any. It returns the volumesnapshotclass and the volumesnapshotcontents as additional items to be backed up.
-func (p *VolumeSnapshotBackupItemAction) Execute(item runtime.Unstructured, backup *velerov1api.Backup) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
+func (p *VolumeSnapshotBackupItemAction) Execute(item runtime.Unstructured, backup *velerov1api.Backup) (runtime.Unstructured, []velero.ResourceIdentifier,
+	string, []velero.ResourceIdentifier, error) {
 	p.Log.Infof("Executing VolumeSnapshotBackupItemAction")
 
+	if backup.Status.Phase == velerov1api.BackupPhaseFinalizing ||
+		backup.Status.Phase == velerov1api.BackupPhaseFinalizingPartiallyFailed {
+		p.Log.WithField("Backup", fmt.Sprintf("%s/%s", backup.Namespace, backup.Name)).
+			WithField("BackupPhase", backup.Status.Phase).
+			Info("Backup is in finalizing phase. Skip the Action.")
+		return item, nil, "", nil, nil
```

Comment:
@Lyndon-Li I see what you mean now about VS deletion during finalize. The deletion code would go right here before the return line.

Reply:
@shubham-pampattiwar This should be fine for OADP too, right? It puts VS deletion in finalize, same phase as in our fork, and with this we could drop the fork commit.

Reply:
And yes, this is cleaner than the current (1.11) way of deleting something in core backup code that was created in a plugin.